From e688eb54fd54d65181e94b854b3568d99cbf3a24 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Tue, 31 Aug 2021 10:37:05 +0200
Subject: [PATCH] build/pack.cc: do not insert payloadflags into .rpm metadata

The flags look like '19T56' where 19 is the compression level
(deterministic), and 56 is the amount of threads (varies from one
host to the next and breaks reproducibility for .rpm).

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>

Rebased to 6.0.2
Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
Signed-off-by: Sumanth Gavini <sumanth.gavini@windriver.com>
---
 build/pack.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/pack.cc b/build/pack.cc
index b1cda6d75..1425d27ba 100644
--- a/build/pack.cc
+++ b/build/pack.cc
@@ -338,7 +338,7 @@ static char *getIOFlags(Package pkg)
 	    headerPutString(pkg->header, RPMTAG_PAYLOADCOMPRESSOR, compr);
 	buf = xstrdup(rpmio_flags);
 	buf[s - rpmio_flags] = '\0';
-	headerPutString(pkg->header, RPMTAG_PAYLOADFLAGS, buf+1);
+	headerPutString(pkg->header, RPMTAG_PAYLOADFLAGS, "");
 	free(buf);
     }
 exit:
